shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/profilePreferences/presentation/editAddress/EditAddressComponent.kt

```diff
@@ -0,0 +1,80 @@
+package com.vml.tutorial.plantshop.profilePreferences.presentation.editAddress
+
+import com.arkivanov.decompose.ComponentContext
+import com.vml.tutorial.plantshop.MR
+import com.vml.tutorial.plantshop.core.presentation.UiText
+import com.vml.tutorial.plantshop.core.utils.componentCoroutineScope
+import com.vml.tutorial.plantshop.profilePreferences.data.ProfileRepository
+import com.vml.tutorial.plantshop.profilePreferences.domain.Address
+import com.vml.tutorial.plantshop.profilePreferences.domain.User
+import com.vml.tutorial.plantshop.profilePreferences.presentation.editAddress.components.EditAddressEvent
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class EditAddressComponent(
+    componentContext: ComponentContext,
+    private val user: User?,
+    private val profileRepository: ProfileRepository,
+    private val onComponentEvent: (event: EditAddressEvent) -> Unit
+) : ComponentContext by componentContext {
+    private val _state = MutableStateFlow(EditAddressState(address = user?.address))
+    val state: StateFlow<EditAddressState> = _state.asStateFlow()
+
+    fun onEvent(event: EditAddressEvent) {
+        when (event) {
+            is EditAddressEvent.StreetNameChanged -> {
+                this._state.update { it.copy(streetName = event.streetName) }
+            }
+
+            is EditAddressEvent.DoorNumberChanged -> {
+                this._state.update { it.copy(doorNumber = event.doorNumber) }
+            }
+
+            is EditAddressEvent.CityChanged -> {
+                this._state.update { it.copy(city = event.city) }
+            }
+
+            is EditAddressEvent.CountryChanged -> {
+                this._state.update { it.copy(country = event.country) }
+            }
+
+            is EditAddressEvent.PostalCodeChanged -> {
+                this._state.update { it.copy(postalCode = event.postalCode) }
+            }
+
+            is EditAddressEvent.AdditionalDescriptionChanged -> {
+                this._state.update { it.copy(additionalDescription = event.additionalDescription) }
+            }
+
+            is EditAddressEvent.SaveClicked -> {
+                user.apply { this?.address = getAddress(state.value) }?.let { updateAddress(it) }
```

Comment:
TODO: show a popup if it was successful. I have a default popup that you can use here if it will be available on this branch.

Reply:
If it is saved successfully, now it shows a snackbar message